stream-chat-android/src/main/java/com/getstream/sdk/chat/view/MessageListViewStyle.kt

```diff
@@ -0,0 +1,574 @@
+                    R.dimen.stream_attach_file_size_text
+                )
+            )
+            .color(
+                R.styleable.MessageListView_streamAttachmentFileSizeTextColorTheirs,
+                ContextCompat.getColor(c, R.color.stream_gray_dark)
+            )
+            .font(
+                R.styleable.MessageListView_streamAttachmentFileSizeTextFontAssetsTheirs,
+                R.styleable.MessageListView_streamAttachmentFileSizeTextFontTheirs
+            )
+            .style(
+                R.styleable.MessageListView_streamAttachmentFileSizeTextStyleTheirs,
+                Typeface.BOLD
+            )
+            .build()
+        attachmentPreviewMaxLines = a.getInt(
+            R.styleable.MessageListView_streamAttachmentPreviewMaxLines,
+            res.getInteger(
+                R.integer.stream_attachment_preview_max_lines
+            )
+        )
+        require(attachmentPreviewMaxLines > 0) { "streamAttachmentPreviewMaxLines value must be greater than 0" }
+
+        // Reaction
+        isReactionEnabled = a.getBoolean(R.styleable.MessageListView_streamReactionEnabled, true)
+        reactionViewBgDrawable =
+            a.getResourceId(R.styleable.MessageListView_streamReactionViewBgDrawable, -1)
+        reactionViewBgColor = a.getColor(
+            R.styleable.MessageListView_streamReactionViewBgColor,
+            ContextCompat.getColor(c, R.color.stream_reaction_input_background)
+        )
+        reactionViewEmojiSize = a.getDimensionPixelSize(
+            R.styleable.MessageListView_streamReactionViewEmojiSize,
+            res.getDimensionPixelSize(
+                R.dimen.stream_reaction_view_emoji_size
+            )
+        )
+        reactionViewEmojiMargin = a.getDimensionPixelSize(
+            R.styleable.MessageListView_streamReactionViewEmojiMargin,
+            res.getDimensionPixelSize(
+                R.dimen.stream_reaction_view_emoji_margin
+            )
+        )
+        // Reaction Dialog
+        reactionInputBgColor = a.getColor(
+            R.styleable.MessageListView_streamReactionInputbgColor,
+            ContextCompat.getColor(c, R.color.stream_reaction_input_background)
+        )
+        reactionInputEmojiSize = a.getDimensionPixelSize(
+            R.styleable.MessageListView_streamReactionInputEmojiSize,
+            res.getDimensionPixelSize(
+                R.dimen.stream_reaction_input_emoji_size
+            )
+        )
+        reactionInputEmojiMargin = a.getDimensionPixelSize(
+            R.styleable.MessageListView_streamReactionInputEmojiMargin,
+            res.getDimensionPixelSize(
+                R.dimen.stream_reaction_input_emoji_margin
+            )
+        )
+
+        // Avatar
+        avatarStyle.avatarWidth = a.getDimensionPixelSize(
+            R.styleable.MessageListView_streamAvatarWidth,
+            res.getDimensionPixelSize(
+                R.dimen.stream_message_avatar_width
+            )
+        )
```

Comment:
Could we move the logic for these nested styles into their own classes? E.g. something like `AvatarStyle.from(a)` here. I guess we'd also have to pass in the specific styleable IDs, so maybe `AvatarStyle.create(a, R.styleable.MLV_steramAvatarWidth, ...)`